api/app/Listeners/ComputeFinalDecisionAndCurrentStep.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Enums\AssessmentDecision;
+use App\Events\AssessmentResultSaved;
+
+class ComputeFinalDecisionAndCurrentStep
+{
+    /**
+     * Create the event listener.
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Handle the event.
+     */
+    public function handle(AssessmentResultSaved $event): void
+    {
+        $result = $event->result;
+        $result->load([
+            'poolCandidate',
+            'poolCandidate.pool.assessmentSteps',
+        ]);
+
+        $candidate = $result->poolCandidate;
+
+        $decisionsPerStep = $candidate->computeDecisionsPerAssessmentStep();
+
+        $currentStep = 1;
```

Comment:
Are we not zero-indexing? 😆 😕 😖

Reply:
Wait, I can't tell if this is a joke or not :sweat_smile: Do you want me to start at 0 ? :stuck_out_tongue: 

I did 1 since that is how sort order words as well and it is easier for comparison.

Reply:
It started as a joke, and then I realized I was actually confused about what to do here. On the one hand, my default assumption about any array-index-like number in code is that it starts from 0, so not doing that could be confusing and lead to misuse. On the other hand, matching sort order makes sense.

(Why didn't we start sort order from 0?)

I guess leave it starting from 1, but leave a comment in the schema that it starts from 1 not 0?

Reply:
Alright, hopefully this is sufficient in communicating that.

[document current step in schema](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/b83552a4213c25170d16fbff522655cb6be975b2)